extension/src/lib/extensionHolder.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2016 RtF-US
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+/* eslint-env node */
+/* global fluid */
+
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+var chrome = chrome || require("sinon-chrome");
+
+fluid.defaults("gpii.chrome.extensionHolder", {
+    gradeNames: "fluid.modelComponent",
+    extensionId: "",
+    members: {
+        extensionInstance: null
+    },
+    events: {
+        onError: null
+    },
+    model: {
+        extensionEnabled: undefined
+    },
+    invokers: {
+        switchStatus: {
+            funcName: "gpii.chrome.extensionHolder.switchStatus",
+            args: "{that}"
+        },
+        setup: {
+            funcName: "gpii.chrome.extensionHolder.setup",
+            args: ["{that}", "{arguments}.0"]
+        }
+    },
+    listeners: {
+        onCreate: {
```

Comment:
It's good practice to namespace listeners. For example this one could change to "onCreate.populate". Namespacing allows you to override an event in another grade, and also provides more control for ordering events.

http://docs.fluidproject.org/infusion/development/InfusionEventSystem.html#namespaced-listeners